fix #19

lolegrand opened the pull request:
Note : I added support for inline content on Clickable text.

boswelja commented:
Sorry, I forgot to assign myself to the issue :sweat_smile: 
Twas fixed just now by https://github.com/boswelja/compose-markdown/pull/21

boswelja commented:
Actually if you want to rebase it, `LocalUriHandler` seems like a much better solution!

lolegrand commented:
Do you want to keep your onClick event bubbleing up to the document or should I remove the onClickLink event.

Note : Maybe we should support the navigation in app. Like if a markdown document is loaded in a view and some links allow user to navigate to other screen.

boswelja commented:
> Do you want to keep your onClick event bubbleing up to the document or should I remove the onClickLink event ?
> 
> Note : Maybe we should support the navigation in app. Like if a markdown document is loaded in a view and some links allow user to navigate to other screen.

I think keeping the `onLinkClick` event is a good idea, it'd also let users customize the link navigation behavior like you said

lolegrand commented:
I have done the modification.

You can merge if it suite you.

boswelja commented on material3/src/androidMain/kotlin/com/boswelja/markdown/material3/UrlLauncher.kt:
```diff
@@ -1,24 +1,20 @@
+import androidx.compose.ui.platform.UriHandler
 
 @Composable
 public actual fun rememberUrlLauncher(): UrlLauncher {
```
We can probably just ditch the whole UrlLauncher thing, what do you think?

lolegrand replied:
Yes, we can. I leave it here because I was thinking about handling deeplLink to navigate in another app.

But apparently it's handled the same way as http URI.

boswelja reviewed (changes requested):
Please also remove the androidx-browser dependency

lolegrand commented:
As you said I deleted the Url abstraction part.

I also remarqued that the event wasn't bubbling up to the material3 Markdown document. 

I added the event and let the choice to the user to override the handler. 

What do you think of it ?

boswelja reviewed (approved):
Looks good to me, thanks!

lolegrand commented:
Thanks !

I don't think I have the right to merge it. Anyhow, I will let you merge it with your next release.

Have a great week 👍
